scripts/ci-e2e.sh

```diff
@@ -143,25 +145,29 @@ cleanup() {
   (gcloud compute instances list --project "$GCP_PROJECT" | grep capg-e2e \
        | awk '{print "gcloud compute instances delete --project '"$GCP_PROJECT"' --quiet " $1 " --zone " $2 "\n"}' \
        | bash) || true
-  (gcloud compute instance-groups list --project "$GCP_PROJECT" | grep capg-e2e \
-       | awk '{print "gcloud compute instance-groups unmanaged delete --project '"$GCP_PROJECT"' --quiet " $1 " --zone " $2 "\n"}' \
-       | bash) || true
   (gcloud compute firewall-rules list --project "$GCP_PROJECT" | grep capg-e2e \
        | awk '{print "gcloud compute firewall-rules delete --project '"$GCP_PROJECT"' --quiet " $1 "\n"}' \
        | bash) || true
+  (gcloud compute instance-groups list --project "$GCP_PROJECT" | grep capg-e2e \
+       | awk '{print "gcloud compute instance-groups unmanaged delete --project '"$GCP_PROJECT"' --quiet " $1 " --zone " $2 "\n"}' \
+       | bash) || true
 
-  # cleanup the networks
-  gcloud compute routers nats delete "${TEST_NAME}-mynat" --project="${GCP_PROJECT}" \
-    --router-region="${GCP_REGION}" --router="${TEST_NAME}-myrouter" --quiet || true
-  gcloud compute routers delete "${TEST_NAME}-myrouter" --project="${GCP_PROJECT}" \
-    --region="${GCP_REGION}" --quiet || true
-
-  if [[ ${GCP_NETWORK_NAME} != "default" ]]; then
-    (gcloud compute firewall-rules list --project "$GCP_PROJECT" | grep "$GCP_NETWORK_NAME" \
-         | awk '{print "gcloud compute firewall-rules delete --project '"$GCP_PROJECT"' --quiet " $1 "\n"}' \
-         | bash) || true
-    gcloud compute networks delete --project="${GCP_PROJECT}" \
-      --quiet "${GCP_NETWORK_NAME}" || true
+  if [[ -n "${SKIP_INIT_NETWORK:-}" ]]; then
+    echo "Skipping GCP network deletion..."
+  else
+    # cleanup the networks
+    gcloud compute routers nats delete "${TEST_NAME}-mynat" --project="${GCP_PROJECT}" \
+      --router-region="${GCP_REGION}" --router="${TEST_NAME}-myrouter" --quiet || true
+    gcloud compute routers delete "${TEST_NAME}-myrouter" --project="${GCP_PROJECT}" \
+      --region="${GCP_REGION}" --quiet || true
+
+    if [[ ${GCP_NETWORK_NAME} != "default" ]]; then
+      (gcloud compute firewall-rules list --project "$GCP_PROJECT" | grep "$GCP_NETWORK_NAME" \
+           | awk '{print "gcloud compute firewall-rules delete --project '"$GCP_PROJECT"' --quiet " $1 "\n"}' \
+           | bash) || true
+      gcloud compute networks delete --project="${GCP_PROJECT}" \
+        --quiet "${GCP_NETWORK_NAME}" || true
+    fi
```

Comment:
The GCP_SERVICE_ACCOUNT is where a lot of these resources such as image will be created during a shared vpc install. Trying to think of a way to incorporate that here on cleanup.